zerver/tests/test_users.py

```diff
@@ -3363,15 +3418,9 @@ def test_do_delete_user(self) -> None:
         self.assertEqual(hamlet.is_active, False)
         self.assertEqual(hamlet.date_joined, hamlet_date_joined)
 
-        # Messages that were sent TO the user are preserved. They'll just appear
-        # to the senders as sent to a deleted dummy user.
-        self.assertEqual(
-            Message.objects.filter(id__in=personal_message_ids_to_hamlet).count(),
-            personal_message_ids_to_hamlet_original_count,
-        )
-
         # Group direct messages from hamlet should have been deleted, but messages of other
         # participants should be kept.
+        self.assertEqual(Message.objects.filter(id__in=personal_message_ids_to_hamlet).count(), 1)
```

Comment:
@timabbott  I think this is a new behaviour for `Personal` messages, right? https://github.com/zulip/zulip/pull/35477

Could you please clarify what should happen to a user’s 1:1 messages versus their group messages when the user is deleted?